lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/openturns/ProjectionStrategyImplementation.hxx

```diff
@@ -60,19 +60,27 @@ public:
   ProjectionStrategyImplementation();
 
   /** Parameter constructor */
-  ProjectionStrategyImplementation(const Distribution & measure);
+  ProjectionStrategyImplementation(const Distribution & measure, 
+                                   const Bool isLeastSquares,
+                                   const Bool isModelSelection);
 
   /** Parameter constructor */
-  ProjectionStrategyImplementation(const WeightedExperiment & weightedExperiment);
+  ProjectionStrategyImplementation(const WeightedExperiment & weightedExperiment, 
+                                   const Bool isLeastSquares,
+                                   const Bool isModelSelection);
 
   /** Parameter constructor */
   ProjectionStrategyImplementation(const Distribution & measure,
-                                   const WeightedExperiment & weightedExperiment);
+                                   const WeightedExperiment & weightedExperiment, 
+                                   const Bool isLeastSquares,
+                                   const Bool isModelSelection);
 
   /** Parameter constructor */
   ProjectionStrategyImplementation(const Sample & inputSample,
                                    const Point & weights,
-                                   const Sample & outputSample);
+                                   const Sample & outputSample, 
+                                   const Bool isLeastSquares,
+                                   const Bool isModelSelection);
```

Comment:
I dont like flags in ctors because it modifies every combination, maybe we could just set them via accessors

Reply:
Having a dedicated method is more robust.

Reply:
Ok. Fixed at dbd305778101e61a9c20a2f5582e9602a8a3bbab.